### PR TITLE
Add support for assets product

### PR DESCRIPTION
**Pull request by dsfish**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,5 +6,6 @@
 /pkg/
 /spec/reports/
 /tmp/
+/vendor/
```
Comment (be9) [inline .gitignore]: Why this?

File: README.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ Or install it yourself as:
 
     $ gem install plaid
 
-The gem supports Ruby 2.1+ only.
+The gem supports Ruby 2.2+ only.
```
Comment (be9) [inline README.md]: Any specific 2.2+ features used here?

File: lib/plaid/models.rb [hunk 3]
```diff
@@ -736,5 +767,325 @@ class Transaction < BaseModel
...
+      property :primary
+    end
+
+    # Public: A representation of an asset report owner.
+    class AssetReportOwner < BaseModel
+      ##
+      # :attr_reader:
+      # Public: A list of names associated with the account by the financial
+      # institution.
+      property :names
```
Comment (be9) [inline lib/plaid/models.rb]: Is it a list of strings here?

File: lib/plaid/models.rb [hunk 4]
```diff
@@ -736,5 +767,325 @@ class Transaction < BaseModel
...
+      ##
+      # :attr_reader:
+      # Public: The settled dollar value; positive values when money moves out
+      # of the account, negative values when money moves in.
+      property :amount
+
+      ##
+      # :attr_reader:
+      # Public: The ISO currency code of the transaction, either USD or CAD.
+      # Always null if unofficial_currency_code is non-null.
```
Comment (be9) [inline lib/plaid/models.rb]: In Ruby speak this would sound as "Always nil if ...... non-nil".

File: lib/plaid/models.rb [hunk 5]
```diff
@@ -736,5 +767,325 @@ class Transaction < BaseModel
...
+      property :iso_currency_code
+
+      ##
+      # :attr_reader:
+      # Public: The unofficial currency code associated with the transaction.
+      # Always null if iso_currency_code is non-null.
```
Comment (be9) [inline lib/plaid/models.rb]: Ditto (null -> nil).

File: lib/plaid/models.rb [hunk 6]
```diff
@@ -736,5 +767,325 @@ class Transaction < BaseModel
...
+      property :middle_name
+
+      ##
+      # :attr_reader:
+      # Public: The user's last name.
+      property :last_name
+
+      ##
+      # :attr_reader:
+      # Public: The user's social security number Format:
```
Comment (be9) [inline lib/plaid/models.rb]: A period missed before "Format"?

File: test/test_asset_report.rb [hunk 7]
```diff
@@ -0,0 +1,66 @@
+require_relative 'test_helper'
+
+# Internal: The test for Plaid::AssetReport.
+class PlaidAssetReportTest < PlaidTest
+  # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+  def test_full_flow
+    # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
```
Comment (be9) [inline test/test_asset_report.rb]: Why do you disable and then enable?

File: test/test_asset_report.rb [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      begin
+        response = @client.asset_report.get(asset_report_token)
+        break
+      rescue Plaid::PlaidAPIError => e
+        raise e if e.error_code != 'PRODUCT_NOT_READY'
+        raise 'Timed out while waiting for asset report generation' \
+          if retries.zero?
+        sleep 1
+      end
+    end
```
Comment (be9) [inline test/test_asset_report.rb]: You don't decrement `retries`. I'd suggest something like this:

```ruby
response = nil

20.times do
  begin
    response = @client.asset_report.get(asset_report_token)
    break     
  rescue Plaid::PlaidAPIError => e
    raise e if e.error_code != 'PRODUCT_NOT_READY'
    sleep 1    
  end
end

assert response, 'Timed out while waiting for asset report generation'
```

File: .gitignore (hunk 1 above)
Comment (dsfish) [reply .gitignore]: I didn't want to commit it along with code changes. Is there a reason to remove this?

File: test/test_asset_report.rb (hunk 7 above)
Comment (dsfish) [reply test/test_asset_report.rb]: I just want to disable the linter for this one method. I can add it as an inline comment instead.

File: README.md (hunk 2 above)
Comment (dsfish) [reply README.md]: It's required by a dependency:

```rdoc-6.0.4 requires ruby version >= 2.2.2, which is incompatible with the current version, ruby 2.1.10p492```

Is it OK to upgrade to 2.2?

File: lib/plaid/models.rb (hunk 3 above)
Comment (dsfish) [reply lib/plaid/models.rb]: :zap

File: lib/plaid/models.rb (hunk 4 above)
Comment (dsfish) [reply lib/plaid/models.rb]: ⚡️

File: lib/plaid/models.rb (hunk 5 above)
Comment (dsfish) [reply lib/plaid/models.rb]: ⚡️

File: lib/plaid/models.rb (hunk 6 above)
Comment (dsfish) [reply lib/plaid/models.rb]: ⚡️

File: test/test_asset_report.rb (hunk 8 above)
Comment (dsfish) [reply test/test_asset_report.rb]: ⚡️

File: .gitignore (hunk 1 above)
Comment (be9) [reply .gitignore]: You can always use `.git/info/exclude` if you need specific stuff to be ignored.

File: README.md (hunk 2 above)
Comment (be9) [reply README.md]: ah I see. Then it's ok.

**Pull request merged**